diff --git a/dom/webidl/Navigator.webidl b/dom/webidl/Navigator.webidl
index d586dc5e4388..9138d90f6b0e 100644
--- a/dom/webidl/Navigator.webidl
+++ b/dom/webidl/Navigator.webidl
@@ -48,7 +48,7 @@ interface mixin NavigatorID {
   readonly attribute DOMString appCodeName; // constant "Mozilla"
   [Constant, Cached]
   readonly attribute DOMString appName; // constant "Netscape"
-  [Constant, Cached, Throws, NeedsCallerType]
+  [Pure, Throws, NeedsCallerType]
   readonly attribute DOMString appVersion;
   [Pure, Cached, Throws, NeedsCallerType]
   readonly attribute DOMString platform;
@@ -175,7 +175,7 @@ partial interface Navigator {
 };
 
 partial interface Navigator {
-  [Throws, Constant, Cached, NeedsCallerType]
+  [Throws, Pure, NeedsCallerType]
   readonly attribute DOMString oscpu;
   // WebKit/Blink support this; Trident/Presto do not.
   readonly attribute DOMString vendor;
@@ -186,4 +186,4 @@ partial interface Navigator {
   // WebKit/Blink/Trident/Presto support this.
   readonly attribute boolean cookieEnabled;
-  [Throws, Constant, Cached, NeedsCallerType]
+  [Throws, Pure, NeedsCallerType]
   readonly attribute DOMString buildID;
